plugins/10_Dragonflight/common.lua

```diff
@@ -512,3 +516,416 @@ local Safari = Class('Safari', Collectible,
+        ELEMENTAL_STORM_PET_REWARDS['sandstorm'], -- Echo of the Cave
+        ELEMENTAL_STORM_PET_REWARDS['firestorm'], -- Echo of the Inferno
+        ELEMENTAL_STORM_PET_REWARDS['snowstorm'], -- Echo of the Depths
+        Spacer(), ELEMENTAL_STORM_FORMULA_REWARDS['all'], -- Design: Elemental Lariat
+        ELEMENTAL_STORM_FORMULA_REWARDS['thunderstorm'], -- Formula: Illusion: Primal Air
+        ELEMENTAL_STORM_FORMULA_REWARDS['sandstorm'], -- Formula: Illusion: Primal Earth
+        ELEMENTAL_STORM_FORMULA_REWARDS['firestorm'], -- Formula: Illusion: Primal Fire
+        ELEMENTAL_STORM_FORMULA_REWARDS['snowstorm'] -- Formula: Illusion: Primal Frost
+    }
+end
+
+ns.node.ElementalStorm = ElementalStorm
+
+-- SPECIFIC ELEMENTAL STORM TOOLTIP EDITOR ------------------------------------
+--
+-- This code will add specific rewards to the tooltip of an Elemental Storm
+-- that is currently happening.
+--
+-- It will show zone-specific AND storm-specific achievements, pets, and recipe
+-- rewards.
+--
+-- General elemental storm nodes will never display at the same location of a
+-- storm that is currently happening (see above).
+--
+-- Additionally, if the Elemental Storms group checkbox is unchecked in the
+-- World Map button then the tooltip will not be affected. This is an
+-- improvement from my Shadowlands integration and should be used for Sentinax
+-- locations in Legion.
+
+local elementalStormHandled = false
+
+hooksecurefunc(GameTooltip, 'Show', function(self)
+    if elementalStormHandled then return end
+    local owner = self:GetOwner()
+    if owner and owner.areaPoiID then
+        if ELEMENTAL_STORM_AREA_POIS[owner.areaPoiID] ~= nil then
+            local mapID = ELEMENTAL_STORM_AREA_POIS[owner.areaPoiID].mapID
+            local type = ELEMENTAL_STORM_AREA_POIS[owner.areaPoiID].type
+            if ns.groups.ELEMENTAL_STORM:GetDisplay(ns.maps[mapID]) then
+                local rewards = {
+                    ELEMENTAL_STORM_MOB_ACHIVEMENTS['all'], -- Elemental Overload
+                    ELEMENTAL_STORM_MOB_ACHIVEMENTS[mapID][type], -- (Example: Thunderstorms in Thaldraszus)
+                    ELEMENTAL_STORM_BOSS_ACHIEVEMENTS[type], -- Stormed Off (Storm type only)
+                    Spacer(), ELEMENTAL_STORM_PET_REWARDS[type], -- Echo of the...
+                    ELEMENTAL_STORM_FORMULA_REWARDS['all'], -- Design: Elemental Lariat
+                    ELEMENTAL_STORM_FORMULA_REWARDS[type] -- Formula: Illusion Primal...
+                }
+                self:AddLine(' ') -- add blank line before rewards
+                for i, reward in ipairs(rewards) do
+                    if reward:IsEnabled() then
+                        reward:Render(self)
+                    end
+                end
+                elementalStormHandled = true
+                self:Show()
+            end
+        end
+    end
+end)
+
+hooksecurefunc(GameTooltip, 'ClearLines',
+    function(self) elementalStormHandled = false end)
```

Comment:
I would use the same hooks we used for the prepatch (the techniques for which I saw in the idTipCommunity addon):

```lua
-- Hook an AreaPOI tooltip
hooksecurefunc(AreaPOIPinMixin, 'TryShowTooltip', function(self)
    if self and self.areaPoiID then
        local mapID = self:GetMap().mapID
        ...
    end
end)

-- Hook a Vignette tooltip
hooksecurefunc(VignettePinMixin, 'OnMouseEnter', function(self)
    if self and self.vignetteInfo then
        ...
    end
end)
```

That should eliminate the need for an `elementalStormHandled` variable.